metadata-io/src/main/java/com/linkedin/metadata/search/elasticsearch/query/request/SearchRequestHandler.java

```diff
@@ -288,6 +293,12 @@ private List<MatchedField> extractMatchedFields(@Nonnull Map<String, HighlightFi
         highlightedFieldNamesAndValues.get(fieldName.get()).add(fieldValue.string());
       }
     }
+    // fallback matched query, non-analyzed field
+    for (String queryName : hit.getMatchedQueries()) {
```

Comment:
What is the purpose of this change? I'm not familiar enough

Reply:
For exact match, keyword fields + term queries are not analyzed, therefore they do not return highlighted fields. This is to populate which field is a matching when the field is unanalyzed. Let me see if I can populate the value. The queries I've seen that hit this were only returning the urn, however if the hit does have the matched field, it could be used to fill the value in.